DotNetHowToIndexers/Program.cs

```diff
@@ -0,0 +1,325 @@
+            DataSource azureCosmosDbDataSource = CreateAzureCosmosDbDataSource(serviceClient, configuration);
+
+            Console.WriteLine("Deleting Cosmos DB indexer...");
+            DeleteIndexerIfExists(serviceClient, AzureCosmosDbIndexerName);
+
+            Console.WriteLine("Creating Cosmos DB indexer...");
+            Indexer azureCosmosDbIndexer = CreateIndexer(serviceClient, AzureCosmosDbDataSourceName, AzureCosmosDbIndexerName);
+
+            return azureCosmosDbIndexer.Name;
+        }
+
+        /// <summary>
+        /// Creates the SearchServiceClient used to make API requests against
+        /// your Azure Search service.
+        /// </summary>
+        private static SearchServiceClient CreateSearchServiceClient(IConfigurationRoot configuration)
+        {
+            string searchServiceName = configuration["SearchServiceName"];
+            string adminApiKey = configuration["SearchServiceAdminApiKey"];
+
+            SearchServiceClient serviceClient = new SearchServiceClient(searchServiceName, new SearchCredentials(adminApiKey));
+            return serviceClient;
+        }
+
+        /// <summary>
+        /// Delete an Azure Search index for the Hotel data type, if it exists.
+        /// </summary>
+        private static void DeleteIndexIfExists(SearchServiceClient serviceClient)
+        {
+            if (serviceClient.Indexes.Exists(IndexName))
+            {
+                serviceClient.Indexes.Delete(IndexName);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Azure Search index for the Hotel data type.
+        /// </summary>
+        private static void CreateIndex(SearchServiceClient serviceClient)
+        {
+            var definition = new Index()
+            {
+                Name = IndexName,
+                Fields = FieldBuilder.BuildForType<Hotel>()
+            };
+
+            serviceClient.Indexes.Create(definition);
+        }
+
+        /// <summary>
+        /// Deletes an Azure Search data source with the given name, if it exists.
+        /// </summary>
+        private static void DeleteDataSourceIfExists(SearchServiceClient serviceClient, string dataSourceName)
+        {
+            if (serviceClient.DataSources.Exists(dataSourceName))
+            {
+                serviceClient.DataSources.Delete(dataSourceName);
+            }
+        }
+
+        /// <summary>
+        /// Creates an Azure Search indexer for the given data source with the given name.
+        /// </summary>
+        private static Indexer CreateIndexer(SearchServiceClient serviceClient, string dataSourceName, string indexerName)
+        {
+            Indexer indexer = new Indexer
+            {
+                DataSourceName = dataSourceName,
+                Name = indexerName,
+                Schedule = IndexerSchedule,
+                TargetIndexName = IndexName
+            };
+            return serviceClient.Indexers.Create(indexer);
```

Comment:
This should be in a try or handle the case that the indexer does not exist (which it will not for the first run).

Reply:
Actually now that, I look at this, I don't see a call to CreateIndex in SetupAzureSqlIndexer() which I would expect we need before this.

Reply:
It doesn't appear to matter whether or not the index is created before the indexer. Should I be calling CreateIndex in all the SetupIndexer methods?

Reply:
I am pretty sure you need to have an index created before you create an indexer.  Are you not getting an error if that is not the case?

Reply:
I see what you mean, I was probably hitting a weird case. Will update this.

Reply:
@liamca If you look at the first few lines of this class, you'll see that the index is created.... Is that something you might've missed? It seems like the index definition is the same for all the different data sources from which we can pull data.

Are you suggesting we have different index definitions for each data source? I believe that this example is functionally correct.